WordPress/gutenberg
# Mobile - RichText/AztecView - Enable customizing the font size for the Heading and Preformatted block.

- Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4414
- Aztec iOS PR: https://github.com/wordpress-mobile/AztecEditor-iOS/pull/1345
- WordPress iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/17707

## Description
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/4112

This PR allows customizing the font size for the Heading and Preformatted block on iOS. It depends on some changes in Aztec iOS, that introduces a new configuration variable `useDefaultFont`.

- It disables font size customization for non-block-based themes.
- Introduces a new prop within RichText for AztecView `blockUseDefaultFont` (iOS only) that will enable the Aztec `useDefaultFont` configuration flag.
- It includes the HTML `tag` within the `text` prop attributes needed to [set the monospace](https://github.com/WordPress/gutenberg/pull/37663/files#diff-3523b4a5a519e3975d67f8a202df9e8f91b3e3aff0168c86598c9e2abcbdf890R709) font for the Preformatted block. We need to pass this data with the content because it has to be enabled before setting the HTML content.

**NOTE**: This feature is still under the `__DEV__` flag, so it's only available in development until is ready to ship.

## How has this been tested?

**Using the build on the WordPress iOS PR**

## Test case 1

**Precondition**: Using a standard theme, e.g. Blank canvas

- Add a Paragraph block
- Open the block settings
- **Expect** to **not see** the font size/line-height picker

## Test case 2

**Precondition**: Using a block-based theme.

- Add a Heading block
- Type some text
- Open the block settings
- Expect to see the Font size picker
- Change the font size
- **Expect** to see the new font size applied
- Change the font size again and set it to `default`
- **Expect** to see the default font size for the selected level (h1, h2, etc)

## Test case 3

**Precondition**: Using a block-based theme.

- Add a Preformatted block
- Type some text
- Open the block settings
- Change the font size of the block
- **Expect** to see the new font size applied

## Screenshots 
Test case 1 | Test case 2 | Test case 3
-|-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/147938134-16e8fee4-0eb8-4f5a-8b11-5d5782d6fc9c.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/147938167-27ec671b-a30a-4f20-9517-4658ab54511b.gif" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/147938213-98550a07-66de-4763-84b2-bc7aaa2ac32b.gif" width="200" /></kbd>

## Types of changes
New feature

## Checklist:
